### ISSUE TITLE
multiChar2fac: check if function actually makes sense for a variable?

### ISSUE THREAD

**Issue by weigtnik** [beckerbenj/eatGADS#120]
This came up through a misunderstanding during the data preparation for FoeDisEs. When called on a variable without any actual character values to recode into a factor, `multiChar2fac` does not give an informative error itself but rather gets terminated by an error from `data.frame` in line [92](https://github.com/beckerbenj/eatGADS/blob/82ff30539a2771e52b41e272e56ce2ee8908b891/R/multiChar2fac.R#L92) which gets an empty character vector `all_levels_sorted` and a numeric vector containing a 1 from `seq_but_skip(to = 0, skip = c(0, 1))` (in this specific case.)

Should the function check if applying it to the given variable actually makes sense - and throw an error if it doesn't?

``` r
library(eatGADS)
df <- data.frame(id = 1:10,
                 somevar = round(runif(10)))
gads <- import_DF(df)
gads <- changeValLabels(gads, "somevar", 0:1, c("unchecked", "checked"))
gads2 <- multiChar2fac(gads, "somevar")
#> Error in data.frame(value = all_levels_sorted, value_new = seq_but_skip(to = length(all_levels_sorted), : Argumente implizieren unterschiedliche Anzahl Zeilen: 0, 1
```

<sup>Created on 2024-12-11 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by beckerbenj** [comment]
I agree, this should throw an informative error similary to what the function does for variables which contain only `NAs`.
